src/Util/BatchRequest.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Http\Client\Util;
+
+use Http\Client\BatchResult;
+use Http\Client\Exception;
+use Http\Client\Exception\BatchException;
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Implements sending multiple request for client not supporting parallels requests
+ *
+ * Internally, use the sendRequest method
+ *
+ * Should be used with Http\Client\HttpPsrClient
+ *
+ * @author Joel Wurtz <jwurtz@jolicode.com>
+ */
+trait BatchRequest
+{
+    /**
+     * {@inheritdoc}
+     */
+    abstract public function sendRequest(RequestInterface $request, array $options = []);
+
+    /**
+     * {@inheritdoc}
+     */
+    public function sendRequests(array $requests, array $options = [])
+    {
+        $batchResult    = new BatchResult();
+        $batchException = new BatchException();
+        $batchException->setResult($batchResult);
```

Comment:
Yeah, it is on purpose that you have to add them one by one, because in most cases (and in this as well) processing of results is linear. This way you don't have to mess with array indicies. Alternatively we could have a method in the BatchException which allows an array input with array elements of a Request and an Exception. But that would be more complicated IMO.